vispy/visuals/ellipse.py

```diff
@@ -119,21 +127,16 @@ def num_segments(self, num_segments):
             raise ValueError('EllipseVisual must consist of more than 1 '
                              'segment')
         self._num_segments = num_segments
-        self._update()
-
-    def _update(self):
-        if self._pos is None:
-            return
-        
-        self._generate_vertices(pos=self._pos, radius=self._radius,
-                                start_angle=self._start_angle,
-                                span_angle=self._span_angle,
-                                num_segments=self._num_segments)
-        if not self._color.is_blank:
-            self.mesh.set_data(vertices=self._vertices,
-                               color=self._color.rgba)
-        if not self._border_color.is_blank:
-            self.border.set_data(pos=self._vertices[1:],
-                                 color=self._border_color.rgba)
-        
-        self.update()
+        self._update_vertices()
+
+    def _update_vertices(self):
+        vertices = self._generate_vertices(center=self._center,
+                                           radius=self._radius,
+                                           start_angle=self._start_angle,
+                                           span_angle=self._span_angle,
+                                           num_segments=self._num_segments)
+
+        # to the PolygonVisual, "pos" is the array of vertices
+        # this will trigger the PolygonVisual's _update()
+        # which gets the ball rolling
+        self.pos = vertices
```

Comment:
We avoided this because Polygon uses a very generic (and expensive) method of triangulation. For regular polygon shapes, there is always a more efficient way to triangulate.